packages/generate/src/edgeql-js/generateCastMaps.ts

```diff
@@ -87,7 +102,9 @@ export const generateCastMaps = (params: GeneratorParams) => {
     ]);
 
     const outerCastableTo = casting(outer.id);
-    staticMap.writeln([t`  A extends ${getRef(outer.name)} ?`]);
+    if (!uncastables.has(outer)) {
```

Comment:
All of these ifs feel dirty, but I understand wanting to keep the runtime logic next to compile time. But I'm not sure now if we should deviate since the compile time is changing its logic somewhat.

Reply:
Oh, I'm perfectly happy to make this division more clear _here_ vs in the generated code: we should prioritize the maintenance of this code over the understandability of the generated code. No change needed, but I'm 100% with you on wanting to branch harder here instead of littering `if` statements everywhere.